modules/codegen/src/smithy4s/codegen/Renderer.scala

```diff
@@ -331,10 +331,14 @@ private[codegen] class Renderer(compilationUnit: CompilationUnit) { self =>
       originalName: String,
       fields: List[Field],
       recursive: Boolean,
-      hints: List[Hint]
+      hints: List[Hint],
+      adtParent: Option[String] = None,
+      additionalLines: Lines = Lines.empty
   ): Lines = {
     val decl = line"case class $name(${renderArgs(fields)})"
     val imports = syntaxImport
+    val implct = if (adtParent.isEmpty) "implicit " else ""
```

Comment:
**issue:** the name could be more descriptive about where this implicit ends up, e.g. `schemaImplicit`